### PR TITLE
Update DataGrip supported version

### PR DESCRIPTION
**Pull request by rmloveland**
Fixes DOC-10150

NB. Updated across v23.1, v23.2, v24.1

### CODE REVIEW

File: src/current/_includes/v23.1/misc/tooling.md [hunk 1]
```diff
@@ -66,7 +66,7 @@ For a list of tools supported by the CockroachDB community, see [Third-Party Too
 
 | IDE | Latest tested version | Support level | Tutorial |
 |-----+-----------------------+---------------+----------|
-| [DataGrip](https://www.jetbrains.com/datagrip/) | 2021.1 | Full | N/A
+| [DataGrip](https://www.jetbrains.com/datagrip/) | 2024.1 | Full | N/A
 | [IntelliJ IDEA](https://www.jetbrains.com/idea/) | 2021.1 | Full | [Use IntelliJ IDEA with CockroachDB]({% link {{ page.version.version }}/intellij-idea.md %})
```
Comment (rimadeodhar) [inline src/current/_includes/v23.1/misc/tooling.md:70]: We can also update the IntelliJ IDEA version? cc @dikshant

**Review by rimadeodhar — APPROVED** [review]

File: src/current/_includes/v23.1/misc/tooling.md (hunk 1 above)
Comment (dikshant) [reply src/current/_includes/v23.1/misc/tooling.md:70]: Yep they are effectively the same!

Comment (rmloveland) [reply src/current/_includes/v23.1/misc/tooling.md]: ok great, thanks - i updated them both to 2024.1

**Review by florence-crl — APPROVED** [review]
LGTM!

**Pull request merged**